browser-extension/mv3/src/service-worker/services/scriptRuleHandler.ts

```diff
@@ -17,7 +28,10 @@ export const applyScriptRules = async (tabId: number, frameId: number, url: stri
 
   scriptRules.forEach((scriptRule) => {
     scriptRule.pairs.forEach((scriptRulePair: ScriptRulePair) => {
-      if (matchSourceUrl(scriptRulePair.source, url)) {
+      if (
+        matchSourceUrl(scriptRulePair.source, url) &&
+        matchResourceTypeFilterWithFrameId(scriptRulePair.source.filters, frameId)
+      ) {
```

Comment:
Can we refractor this and use `ruleMatcher.matchRuleWithRequest` instead. This would make sure all the source filter code remains at single place